### ISSUE TITLE
`shuffle` argument when initializing Sampler-related classes in `trainer.py`

### ISSUE THREAD

**Issue by gugarosa** [huggingface/transformers#21888]
### Feature request

Add support for an additional keyword argument (`shuffle`) in `DistributedSampler` (https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L839).

### Motivation

It samples data in a deterministic way across multiple processes. However, when using a contiguous array of pre-encoded data, using a shuffler in the sampler could avoid sequential input_ids for extreme long files and possibly diminish the number of spikes in the loss.

### Your contribution

Would you consider worth in adding it? In my use case, I have to re-write the whole `_get_train_sampler()` just to set `shuffle=True`.

If necessary, I can contribute with a PR.

Thanks for your attention and best regards,
Gustavo.

**Comment by sgugger** [comment]
`shuffle=True` is the default for this DistributedSampler (see [doc](https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler)). There is no need to pass it.

**Comment by gugarosa** [comment]
You are right, my bad. I overlooked the documentation.
